### DISCUSSION TITLE
Starting Dynaframe from command line

### DISCUSSION THREAD

**Discussion by UncommonDad** [Geektoolkit/Dynaframe3#73] [Q&A]
I'm pretty new to Linux and raspberry pi.  I'm wondering how to restart Dynaframe once you shut the application down?

Thx!

**Comment by Geektoolkit** [comment]
You'll want to start the terminal, and then change to the Dynaframe directory. Type:
cd Dynaframe 

Then

.\Dynaframe 

And it should start back up!

**Reply by UncommonDad** [reply]
Awesome...on a similar note is there a way to kill the process from the command line?

My wife absolutely loves the frame I made for her....thx!

**Reply by UncommonDad** [reply]
I found a great little app that gives you all kinds of info on your Raspberry Pi.  Gives you process lists, allows you to kill processes, reboot, memory usage, etc.

https://play.google.com/store/apps/details?id=it.Ettore.raspcontroller

**Reply by Geektoolkit** [reply]
whoa! I'll add that to my 'stuff to try' list for my tablet, thanks that looks cool!  For killing a process from the command line I usually type:
sudo pkill Dynaframe 
to get Dynaframe down.

**Comment by dazzign** [comment]
Love the idea around this project, but I am also very new to Linus and Raspberry Pi....I have installed Dynaframe on a Pi 4...Seemed to install fine, but does not launch the web page after reboot....Running the command .\Dynfram also doesn't do anything. I have tried accessing the config page from a web browser on Windows, but I get a connection refused message. I am running a headless setup at the moment...connected to ethernet and accessed through VNC and this seems to work fine....Guessing here, but could it possibly be a port thats not open ?

**Reply by UncommonDad** [reply]
I had the same problem.  You have to connect using the IP address and include :8000 after it.

For example 10.0.0.100:8000

**Reply by dazzign** [reply]
Thanks, yeah I have tried the port number, but no success....I dont think the Dynaframe is actually running at all

**Comment by qwksilver** [comment]
so it's pi 4, lets do some more diagnostics
was it a new installation of raspberry pi os, and if so which source image did you use
what ip address is your frame on.
which version of Dynaframe did you install?
running any command in linux is case sensitive so .\dynaframe  won't work and .\Dynaframe will (or should)

from vnc, open terminal, and do the following commands
``` 
cd Dynaframe
ls -l
```
copy and paste your output, then run
`.\Dynaframe`

and copy and patse any output you get at that point, or describe if any windows happen to open.

**Reply by dazzign** [reply]
Thanks so much for you follow up - I really want to get this going....

I have run the commands and dropped the output into a txt file (attached).....the list gave me a full directory listing of files and the Dynaframe command gave me the following error

pi@raspberrypi:~/Dynaframe $ .\Dynaframe
bash: .Dynaframe: command not found
pi@raspberrypi:~/Dynaframe $ 

[Dynaframe.txt](https://github.com/Geektoolkit/Dynaframe3/files/5924123/Dynaframe.txt)

**Reply by Geektoolkit** [reply]
Ah perfect. Please try:
./Dynaframe 

(Reverse the slash).  That should get you going.

**Reply by dazzign** [reply]
Thanks so much for the fast response....definitely got further, but still not running

pi@raspberrypi:~/Dynaframe $ ./Dynaframe
2/4/2021 4:20 PM:Reload settings was true
2/4/2021 4:20 PM:Refresh Directories was true
2/4/2021 4:20 PM:ERROR: GoToNext was unable to find current item in playlist!
2/4/2021 4:20 PM:New List generated! Contains: 0 items.
2/4/2021 4:20 PM:----------------------Begin Playlist Dump----------------
2/4/2021 4:20 PM:------------------- END PLAYLIST DUMP-------------------
2/4/2021 4:20 PM:New List generated! Contains: 0 items.
2/4/2021 4:20 PM:----------------------Begin Playlist Dump----------------
2/4/2021 4:20 PM:------------------- END PLAYLIST DUMP-------------------
2/4/2021 4:20 PM:Reached end of the list!
2/4/2021 4:20 PM:New List generated! Contains: 0 items.
2/4/2021 4:20 PM:----------------------Begin Playlist Dump----------------
2/4/2021 4:20 PM:------------------- END PLAYLIST DUMP-------------------
2/4/2021 4:20 PM:New List generated! Contains: 0 items.
2/4/2021 4:20 PM:----------------------Begin Playlist Dump----------------
2/4/2021 4:20 PM:------------------- END PLAYLIST DUMP-------------------
Unhandled exception. System.InvalidOperationException: Sequence contains no elements
   at System.Linq.ThrowHelper.ThrowNoElementsException()
   at System.Linq.Enumerable.First[TSource](IEnumerable`1 source)
   at Dynaframe3.PlayListEngine.GoToNext()
   at Dynaframe3.MainWindow.Timer_Tick(Object sender, EventArgs e)
   at Dynaframe3.MainWindow.InitializeComponent()
   at Dynaframe3.MainWindow..ctor()
   at Dynaframe3.App.OnFrameworkInitializationCompleted()
   at Avalonia.Controls.AppBuilderBase`1.Setup()
   at Avalonia.Controls.AppBuilderBase`1.SetupWithLifetime(IApplicationLifetime lifetime)
   at Avalonia.ClassicDesktopStyleApplicationLifetimeExtensions.StartWithClassicDesktopLifetime[T](T builder, String[] args, ShutdownMode shutdownMode)
   at Dynaframe3.Program.Main(String[] args)
Aborted
pi@raspberrypi:~/Dynaframe $

**Comment by qwksilver** [comment]
try this from terminal or ssh
sudo wget https://github.com/Geektoolkit/Dynaframe3/releases/download/2.16/install.sh && bash install.sh; rm -f install.sh; sudo reboot

let us know if it's running after it reboots.  it's looking like it just doesn't have any photos to work with though.
this new version should self correct that issue and give you an upload button in the web interface to add some to start out with
let us know, sorry for the delay.

**Reply by UncommonDad** [reply]
I'm not experiencing the same issue myself, but that will update me to the latest build?

(Kind of particular about keeping current updates...lol)

**Reply by Spinner65** [reply]
You can find the different releases here:

https://github.com/Geektoolkit/Dynaframe3/releases

**Reply by dazzign** [reply]
Thanks Folks...all running now......The web controls for 2.16 are really good

**Reply by Geektoolkit** [reply]
Glad you're up and running!
